src/Std/Data/DHashMap/Lemmas.lean

```diff
@@ -246,9 +246,17 @@ theorem get?_insert_self [LawfulBEq α] {k : α} {v : β k} : (m.insert k v).get
 theorem contains_eq_isSome_get? [LawfulBEq α] {a : α} : m.contains a = (m.get? a).isSome :=
   Raw₀.contains_eq_isSome_get? ⟨m.1, _⟩ m.2
 
+@[simp]
+theorem isSome_get?_eq_contains [LawfulBEq α] {a : α} : (m.get? a).isSome = m.contains a :=
+  contains_eq_isSome_get?.symm
+
 theorem mem_iff_isSome_get? [LawfulBEq α] {a : α} : a ∈ m ↔ (m.get? a).isSome :=
   Bool.eq_iff_iff.mp contains_eq_isSome_get?
 
+@[simp]
+theorem isSome_get?_iff_mem [LawfulBEq α] {a : α} : (m.get? a).isSome ↔ a ∈ m :=
```

Comment:
This will never trigger, because `simp` simplifies smaller terms before considering larger terms, and so `isSome_get?_eq_contains` will fire first. I think the correct solution here is to turn abound `mem_iff_contains` and make `contains_iff_mem` into a `simp` lemma?

Reply:
I've pushed this now.